packages/gestalt/src/SheetConfirmationPopover.js

```diff
@@ -0,0 +1,161 @@
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider.js';
+import { ESCAPE } from './keyCodes.js';
+
+type Props = {|
+  onDismiss: () => void,
+  anchor: ?HTMLElement,
+  dismissConfirmation?:
+    | boolean
+    | {|
+        message?: string,
+        subtext?: string,
+        primaryAction?: {|
+          accessibilityLabel?: string,
+          text?: string,
+          onClick?: ({|
+            event:
+              | SyntheticMouseEvent<HTMLButtonElement>
+              | SyntheticMouseEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLButtonElement>,
+          |}) => void,
+        |},
+        secondaryAction?: {|
+          accessibilityLabel?: string,
+          text?: string,
+          onClick?: ({|
+            event:
+              | SyntheticMouseEvent<HTMLButtonElement>
+              | SyntheticMouseEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLButtonElement>,
+          |}) => void,
+        |},
+      |},
+|};
+
+export default function SheetConfirmationPopover({
+  anchor,
+  dismissConfirmation,
+  onDismiss,
+}: Props): Node {
+  const { onAnimatedDismiss } = useAnimation();
+
+  const {
+    dismissConfirmationMessage: dismissConfirmationMessageDefault,
+    dismissConfirmationSubtext: dismissConfirmationSubtextDefault,
+    dismissConfirmationPrimaryActionText: dismissConfirmationPrimaryActionTextDefault,
+    dismissConfirmationSecondaryActionText: dismissConfirmationSecondaryActionTextDefault,
+    dismissConfirmationPrimaryActionTextLabel: dismissConfirmationPrimaryActionTextLabelDefault,
+    dismissConfirmationSecondaryActionTextLabel: dismissConfirmationSecondaryActionTextLabelDefault,
+  } = useDefaultLabelContext('Sheet');
+
+  const confirmationButtonRef = useRef();
+
+  useEffect(() => {
+    if (confirmationButtonRef.current) {
+      confirmationButtonRef.current.focus();
+    }
+  }, [confirmationButtonRef]);
+
+  // Handle onDismiss triggering from ESC keyup event
+  useEffect(() => {
+    function handleKeyDown(event: {| keyCode: number, stopPropagation: () => void |}) {
+      if (event.keyCode === ESCAPE) {
+        event.stopPropagation();
+      }
+    }
+
+    window.addEventListener('key', handleKeyDown);
+    return function cleanup() {
+      window.removeEventListener('keyup', handleKeyDown);
+    };
+  }, []);
+
+  return (
+    <Popover
+      anchor={anchor}
+      idealDirection="down"
+      onDismiss={() => onDismiss()}
+      positionRelativeToAnchor
+      role="dialog"
+      size="md"
+    >
+      <TrapFocusBehavior>
+        <Box padding={3} width="100%">
+          <Flex direction="column" gap={4}>
+            <Box role="alert">
+              <Flex direction="column" gap={2} alignItems="center" width="100%">
+                <Text weight="bold">
+                  {typeof dismissConfirmation === 'boolean'
```

Comment:
The props changes mentioned above would simplify all this logic as well

Reply:
DONE